### PR TITLE
Examples for Tutorial 2 – Custom UX

### PR DESCRIPTION
**Pull request by BohdanBilonoh**
Need to add examples for Tutorial 2 – Custom UX from [doc](https://docs.google.com/document/d/19cVTBPadBSYWdEf6f933mdOQCrPgMU_VdG1EaTNyb2c/edit)

[JIRA Issue](https://jira.ops.aol.com/browse/OMSDK-520)

### CODE REVIEW

File: NewTutorials/Tutorial/Info.plist [hunk 1]
```diff
@@ -41,5 +41,10 @@
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationLandscapeRight</string>
 	</array>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoads</key>
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/Info.plist]: Why do you need this enabled? - We have https everywhere.

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 2]
```diff
@@ -8,7 +8,13 @@ import PlayerControls
 class PlayerViewControllerWrapper: UIViewController {
     struct Props {
         var player: Future<Result<Player>>?
+        // Using default controls but it's possible to use custom by subclassing from ContentControlsViewController and set it to contentControlsViewController
         var controlsViewController: ContentControlsViewController = DefaultControlsViewController()
+        var controlsColor: UIColor?
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: These fields look like separate struct inside `Props`.

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 3]
```diff
@@ -29,10 +35,43 @@ class PlayerViewControllerWrapper: UIViewController {
...
+                    guard var controls = player.item.playable else { return props }
+                    
+                    // Changing color of live dot indicator
+                    controls.live.dotColor = self?.props.liveDotColor
+                    
+                    // Hiding 10s seek button and setting button
+                    if self?.props.isHiddenSomeControls == true {
+                        controls.seekbar?.seeker.seekTo = nil
+                        controls.settings = .hidden
+                    }
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]:
```
else {
// unhide them?
}```

File: NewTutorials/Tutorial/SetupTutorials.swift [hunk 4]
```diff
@@ -4,24 +4,93 @@ import UIKit
...
-            fatalError("Wrapper not found")
-        }
-        
-        vc.show(viewController: wrapper)
-        wrapper.props.player = player
+
+func bind(player: Future<Result<Player>>,
+          wrapper: PlayerViewControllerWrapper,
+          vc: TutorialCasesViewController) {
+    vc.show(viewController: wrapper)
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/SetupTutorials.swift]: Function `bind` shows view controller, seriously?

File: NewTutorials/Tutorial/TutorialsViewController.swift [hunk 5]
```diff
@@ -2,12 +2,15 @@
...
+
 class TutorialsViewController: UITableViewController {
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         guard let identifier = segue.identifier else { return }
+        guard let vc = segue.destination as? TutorialCasesViewController else { fatalError("Unknown destination") }
         
-        switch (identifier, segue.destination) {
-        case ("PlayingVideos", let vc as TutorialCasesViewController): setupPlayingVideos(vc: vc)
+        switch (identifier) {
+        case ("PlayingVideos"): setupPlayingVideos(vc: vc)
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/TutorialsViewController.swift]: Parenthesis `( )` not needed.

File: NewTutorials/Tutorial/SetupTutorials.swift [hunk 6]
```diff
@@ -4,24 +4,93 @@ import UIKit
...
+
+func playerViewControllerWrapper() -> PlayerViewControllerWrapper {
+    guard let wrapper = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "PlayerViewControllerWrapper") as? PlayerViewControllerWrapper else {
+        fatalError("Wrapper not found")
     }
-    
+    return wrapper
+}
+
+func setupPlayingVideos(vc: TutorialCasesViewController) {
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/SetupTutorials.swift]: `vc` is a bad parameter name

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/SetupTutorials.swift [hunk 7]
```diff
@@ -4,24 +4,95 @@ import UIKit
...
-            fatalError("Wrapper not found")
-        }
-        
-        vc.show(viewController: wrapper)
-        wrapper.props.player = player
+
+func show(wrapper: PlayerViewControllerWrapper,
+          in tutorialCasesViewController: TutorialCasesViewController,
+          with player: Future<Result<Player>>) {
+    tutorialCasesViewController.show(viewController: wrapper)
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/SetupTutorials.swift]: Can wrapper be created inside this function?

Comment (AndriiMoskvin) [reply NewTutorials/Tutorial/SetupTutorials.swift]: No reason to pass it each time to this function.

Comment (AndriiMoskvin) [reply NewTutorials/Tutorial/SetupTutorials.swift]: The same about `TutorialCasesViewContoller` - maybe function `show` should be maid inside this class?

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 8]
```diff
@@ -29,10 +39,43 @@ class PlayerViewControllerWrapper: UIViewController {
                         completion: nil)
             }
             
+            func render(player: Player) {
+                playerViewController?.customizeContentControlsProps = { [weak self] props in
+                    // Mudifing content props only if content video can be played
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: `guard let self = self else {  return props }`

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 9]
```diff
@@ -29,10 +39,44 @@ class PlayerViewControllerWrapper: UIViewController {
                         completion: nil)
             }
             
+            func render(player: Player) {
+                playerViewController?.customizeContentControlsProps = { [weak self] props in
+                    guard let strongSelf = self else { return props }
+                    // Mudifing content props only if content video can be played
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: `Modifying`

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 10]
```diff
@@ -29,10 +39,44 @@ class PlayerViewControllerWrapper: UIViewController {
...
+                    guard var player = props.player else { return props }
+                    guard var controls = player.item.playable else { return props }
+                    
+                    // Changing color of live dot indicator
+                    controls.live.dotColor = strongSelf.props.controls.liveDotColor
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: Can we make each of these snipppet as a function with self-descriptive name?

**Review by AndriiMoskvin — DISMISSED** [review]

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]
☝️

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 11]
```diff
@@ -40,29 +40,36 @@ class PlayerViewControllerWrapper: UIViewController {
             }
             
             func render(player: Player) {
+                typealias Controls = PlayerControls.ContentControlsViewController.Props.Player.Item.Controls
+                
                 playerViewController?.customizeContentControlsProps = { [weak self] props in
                     guard let strongSelf = self else { return props }
-                    // Mudifing content props only if content video can be played
+                    // Mudifying content props only if content video can be played
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: `Modifying`

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift (hunk 11 above)
Comment (BohdanBilonoh) [reply NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: :disappointed_relieved:

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**